dashboard/pages/explain_distsql.tsx

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright 2022 Singularity Data
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+import { Box, Button, Stack, Textarea } from "@chakra-ui/react"
+import { Fragment } from "react"
+import Title from "../components/Title"
+
+import React, { useState } from "react"
+import { Col, Container } from "react-bootstrap"
+import ReactFlow, {
+  Background,
+  Controls,
+  Edge,
+  MiniMap,
+  Node,
+} from "react-flow-renderer"
+import styled from "styled-components"
+import NodeType from "./node"
+
+import * as d3 from "d3"
+
+const ContainerDiv = styled(Container)`
+  font-family: sans-serif;
+  text-align: left;
+`
+
+const DemoArea = styled(Col)`
+  width: 100%;
+  height: 80vh;
+`
+
+const position = {
+  x: 200,
+  y: 100,
+}
+
+const nodeTypes = { node: NodeType }
+
+function getColor() {
+  return (
+    "hsl(" +
+    360 * Math.random() +
+    "," +
+    (25 + 70 * Math.random()) +
+    "%," +
+    (85 + 10 * Math.random()) +
+    "%)"
+  )
+}
+
+function getStyle() {
+  return {
+    background: getColor(),
+    height: 50,
+    width: 200,
+    border: "0.5px solid black",
+    padding: "5px",
+    "border-radius": "5px",
+  }
+}
+
+function layoutElements(nodeList: any, edgeList: any) {
+  const idToNode = new Map()
+  nodeList.forEach((node: { id: any }) => {
+    idToNode.set(node.id, [{ id: node.id, children: [] }, node])
+  })
+
+  edgeList.forEach((edge: any) => {
+    const sourceNode = idToNode.get(edge.source)[0]
+    const targetNode = idToNode.get(edge.target)[0]
+    sourceNode.children.push(targetNode)
+  })
+
+  var rootNode = idToNode.get(stageToNode[rootStageId].toString())[0]
+  var root = d3.hierarchy(rootNode)
+  var tree = d3.tree().nodeSize([200, 100])
+  const treeRoot = tree(root)
+
+  treeRoot.each((treeNode: { x: number; y: number; data: any }) => {
+    const node = idToNode.get(treeNode.data.id)[1]
+    if (node == undefined) return
+    node["position"] = {
+      x: treeNode.x,
+      y: treeNode.y,
+    }
+  })
+}
+
+var visited = new Set()
+var stageToNode: { [key: string]: number } = {}
+var nodeStagePairs: number[][] = []
+var rootStageId: string = "0"
+
+function parseSubElements(root: any, style: any, nodeList: any, edgeList: any) {
+  if (root["children"].length == 0) return
```

Comment:
We're using TypeScript, we don't need to access everything use brackets. Consider defining an interface for all these elements.